docs/articles/documentation/extending-testcafe/custom-browser-provider-plugin/index.md

```diff
@@ -0,0 +1,200 @@
+---
+layout: docs
+title: Custom Browser Provider Plugin
+permalink: /documentation/extending-testcafe/custom-browser-provider-plugin/
+---
+# Custom Browser Provider Plugin
+
+TestCafe has built-in support for testing in a number of popular browsers (see [Using Browsers](../../using-testcafe/common-concepts/using-browsers.md)).
+If you need to use a different browser (for example, a browser from a cloud service), you can create a **custom browser provider plugin**.
+The provider should expose methods for performing common actions on the browser windows during testing: opening and closing it, taking screenshot, etc.
+
+To create a browser provider plugin, go through the following steps.
```

Comment:
@kirovboris Eeh, why not? I don't get your point. Make your plugin configurable to allow it accept keys and passwords. We suggest npm as a primary plugin distribution system